lib/web_ui/lib/src/engine/pointer_binding/event_position_helper.dart

```diff
@@ -33,6 +43,24 @@ ui.Offset computeEventOffsetToTarget(DomMouseEvent event, DomElement actualTarge
   return ui.Offset(event.offsetX, event.offsetY);
 }
 
+/// Computes the offsets for input nodes, which live outside of the shadowDOM.
+/// Since inputs can be transformed (scaled, translated, etc), we can't rely on
+/// `event.offset` to be accurate. `_computeOffsetRelativeToActualTarget` only
+/// handles the case where inputs are translated, but will have issues for scaled
+/// inputs (see: https://github.com/flutter/flutter/issues/125948).
+///
+/// We compute the offsets here by using the text input geometry data that is
+/// sent from the framework, which includes information on how to transform the
+/// underlying input element. We transform the `event.offset` points we receive
+/// using the values from the input's transform matrix.
+ui.Offset _computeOffsetForInputs(DomMouseEvent event) {
+  final Float32List matrix = textEditing.strategy.geometry!.globalTransform;
+  final FastMatrix32 fastMatrix = FastMatrix32(matrix);
+  fastMatrix.transform(event.offsetX, event.offsetY);
```

Comment:
If this is performance-critical, I'd first check if the matrix is within the limits that `FastMatrix32` can deal with and only use it then, and fallback to full matrix otherwise. I think`transformKindOf` should give you the right info.

Reply:
Why not https://developer.mozilla.org/en-US/docs/Web/API/DOMMatrixReadOnly / DOMMatrix?

Reply:
@ditman `DOMMatrixReadOnly` and its instance method `transformPoint` could certainly work, although it's missing documentation on MDN.  But the API seems well supported across different browsers.  We can also init it with `globalTransform` which should be a 16 length array. Do you think there would be significant advantages to using the browser API? Maybe their `transformPoint` is more optimized?

@yjbanov I don't see any docs outlining the limitations of the `FastMatrix32`, do you mean it only support matrices that perform simple 2D transforms and `Matrix4` should be used for 3D transforms?

Reply:
@yjbanov mentioned that DOMMatrix might not be fully supported everywhere, so I guess we're sticking to our own implementation!